receivers-in-go.md

```diff
@@ -0,0 +1,266 @@
+to a type, the receiver is the instance of the type for which it was called.
+
+Choosing the name of a receiver is not always a trivial task. Should we be lazy 
+and name them all the same (like `this` or `self`)? Or treat them not unlike 
+local variables by abbreviating the type (like `srv` to a `Server`type )? Or 
+maybe something even more nuanced?
+
+And what are the consequences? How will our code suffer if we choose one 
+approach over the other? Let's explore.
+
+
+## Quick refresher on Go structs
+
+```go
+type Server struct {
+    ...
+}
+
+func (srv *Server) Close() error {
+    ...
+}
+
+func (srv Server) Name() string {
+    ...
+}
+```
+
+Every time I start writing somekind of server, it starts out looking like this.  
+It's a type that holds information about our server, like a `net.Listener` 
+socket, and it has a `Close()` method that shuts down the server. Easy enough.
+
+The *receiver* of the `Close()` method is the `(srv *Server)` part. This says 
+that inside of the `Close()` method declaration, the scope will have a `srv` 
+variable that is a reference to the instance of the `Server` that it's being 
+called on.  That is:
+
+```go
+myServer := &Server{}
+myServer.Close()
+```
+
+In this case, the `srv` that is referenced inside of the `myServer.Close()` is 
+effectively the same variable as `myServer`. They're both references to the same 
+`Server` instance.
+
+
+### Facts about Go methods and receivers
+
+While we can call a method on a type instance and get the receiver implicitly, 
+it can also be called explicitly:
+
+```go
+myServer := &Server{}
+Server.Name(myServer) // same as myServer.Name()
+(*Server).Close(&myServer) // same as myServer.Close()
+```
+
+These functions can be passed around as references just like any other function, 
+with the implicit receiver or without:
+
+```go
+withReceiver := myServer.Name
+without := Server.Name
+```
+
+Receivers can be passed by reference or passed by value.
+
+```go
+func (byValue Server) Hello() { ... }
+func (byReference *Server) Bye() { ... }
+```
+
+This is to illustrate that struct methods in Go are merely thin sugar over 
+traditional C-style struct helper declarations. An equivalent C method might 
+look like this:
+
+```c
+void server_close(server *srv) { ... }
+```
+
+Go helps by namespacing the methods and implicitly passing the receiver when 
+called on an instance, but otherwise there is very little magic going on.
+
+In other languages where `this` and `self` is a thing (Python, Ruby, JavaScript, 
+and so on) it's a much more complicated situation. These are not vanilla local 
+variables wearing fancy pants. The thing we might expect `this` to represent 
+inside of a method could actually represent something very different once 
+inheretance or metaclasses had their way. In effect, it might not make any sense 
+to give contextual names like `srv` rather than `self` in Python, but it 
+definitely makes sense in Go.
```

Comment:
explain? how do i better rephrase? (do you feel this section even adds anything? i kinda wanna tear it out)